### ISSUE TITLE
Speeding up Pint by re-writing its core in Cython

### ISSUE THREAD

**Issue by MatthieuDartiailh** [hgrecco/pint#193]
Hi, 

First thank you for the great library and your awesome work on pyvisa too. I know my title is a bit provocative and I will try to explain my point.
In one of my project ([HQCMeas](https://github.com/MatthieuDartiailh/HQCMeas), which is a framework to control experiments), I would like to include unit handling in my instruments drivers (in a way similar to Lantz but without the dependency on Qt, and with some other features too). My issue is that currently in my lab the tightest loop we have is about 16 ms on average (it includes setting a voltage, getting a lock-in measure, doing some formatting and saving to a file), and I don't want to degrade the performance of my code by adding unit handling. I did some benchmarking of pint and found that it took about 800 µs just to convert mV to V which is definitively too long for me.
I am hence willing to try to rewrite the core of Pint in cython to speed it up (I would ideally love to get a factor 100 or 1000 but I don't know if it will be possible). To allow users to still use pint as a pure python module, it may be best to create a new project (cPint ?) with the same API that pint could use when present for all low-levels tasks.
As you are the project leader, I would like to know your opinion. And if you could give me any hints on what I should focus (I guess UnitRegistry and Quantity are the two main things to work on) it would be greatly appreciated.
Best regards

Matthieu Dartiailh

**Comment by hgrecco** [comment]
I am moving forward with #215 and might require that we reorganize part of the code. What is your status with this?

**Comment by MatthieuDartiailh** [comment]
I have been quite busy with other stuff lately but I hope to finsh the Unit part soon. Actually I just pushed a commit dealing with a big part of it. I still have some issues with some ufunc and some tests from test_issues which I will have to re-write. I must also add specific tests for the new Unit class. Hopefully I can at least fix everything this week. As we splitted the unit module it would be nice to merge #202 before. I will do my best.
